log10/_httpx_utils.py

```diff
@@ -338,38 +338,53 @@ def parse_anthropic_responses(self, responses: list[str]):
 
     def parse_openai_responses(self, responses: list[str]):
         r_json = None
+        finish_reason = None
+        last_message_choices = None
         for r in responses:
             if self.is_openai_response_end_reached(r):
                 break
 
             # loading the substring of response text after 'data: '.
             # example: 'data: {"choices":[{"text":"Hello, how can I help you today?"}]}'
             r_json = json.loads(r[6:])
-            delta = r_json["choices"][0]["delta"]
-
-            # Delta may have content
-            if "content" in delta:
-                content = delta["content"]
-                if content:
-                    self.full_content += content
-
-            # May be a function call, and have to reconstruct the arguments
-            if "function_call" in delta:
-                # May be function name
-                if "name" in delta["function_call"]:
-                    self.function_name = delta["function_call"]["name"]
-                # May be function arguments
-                if "arguments" in delta["function_call"]:
-                    self.full_argument += delta["function_call"]["arguments"]
-
-            if tc := delta.get("tool_calls", []):
-                if tc[0].get("id", ""):
-                    self.tool_calls.append(tc[0])
-                elif tc[0].get("function", {}).get("arguments", ""):
-                    idx = tc[0].get("index")
-                    self.tool_calls[idx]["function"]["arguments"] += tc[0]["function"]["arguments"]
+
+            if "choices" in r_json and r_json["choices"]:
```

Comment:
`r_json.get("choices", {})` and `get()` generally is a bit more graceful when the key isn't available. Could make the rest of the code more robust/bit more concise